test/e2e/serial/tests/workload_placement.go

```diff
@@ -114,7 +114,12 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 				corev1.ResourceMemory: resource.MustParse("8G"),
 			}
 			By("padding the nodes before test start")
-			err := padder.Nodes(numOfNodeToBePadded).UntilAvailableIsResourceList(rl).Pad(timeout, e2epadder.PaddingOptions{})
```

Comment:
I get the idea of the fix, yet I believe tests that requires nrtcandidates that are >=2 should have been skipped and not reach this code, in this specific case the test should be skipped here:
https://github.com/openshift-kni/numaresources-operator/blob/3abdd3cde6213800f43ceaee6d4603b1a1501e9c/test/e2e/serial/tests/workload_placement.go#L106

Reply:
ah this points to the nrts with the policies. maybe applying similar solution here to what you've done in the second commit

Reply:
true. The problem is that the padder pkg will try to do -earlier than that point- a padding of the candidate nodes until the free resources are what is declared in `rl` above. Thus 3 cpus, 8G of RAM. In some (arguably debatable) CI setup we can have virtual worker nodes mixed with physical worker nodes, and the virtual nodes are usually way weaker than physical worker.

Arguably, albeit that's a pretty weak stance, we should do check the available resources on nodes before we attempt to pad them. This fix only partially address this issue, and incidentally.

Reply:
I managed to improve this in the followup commits, but a overhaul is probably still in order (material for future PRs)